src/components/common/YourCompany/EntityName.vue

```diff
@@ -266,13 +285,27 @@ export default class EntityName extends Mixins(NameRequestMixin) {
   @Getter(useStore) isLimitedRestorationExtension!: boolean
   @Getter(useStore) isNumberedCompany!: boolean
   @Getter(useStore) isSpecialResolutionFiling!: boolean
+  @Getter(useStore) isNameChangedByType!: boolean
 
   // store actions
   @Action(useStore) setBusinessInformation!: ActionBindingIF
   @Action(useStore) setEditingCompanyName!: ActionBindingIF
   @Action(useStore) setNameRequest!: ActionBindingIF
   @Action(useStore) setValidComponent!: ActionBindingIF
 
+  get shouldShowUndoButton () {
+    return (this.hasCompanyNameChanged ||
+            (this.hasBusinessNameChanged &&
+             (this.isAlterationFiling || this.isFirmChangeFiling || this.isSpecialResolutionFiling))) &&
+            !this.isNameChangedByType
+  }
+
+  get shouldShowTypeDetail () {
+    return !this.hasNewNr && this.hasBusinessNameChanged &&
+           (this.isAlterationFiling || this.isFirmChangeFiling || this.isFirmConversionFiling) &&
+           !this.isNameChangedByType
+  }
```

Comment:
I like these 2 getters instead of having the code in the template.

Could you add a comment to each to describe when then are True?

(Also add return types.)